src/fbrs/controller/MarketsController.java

```diff
@@ -30,6 +54,43 @@ public void printAllMarkets() throws IOException {
 
     @Override
     public void initialize(URL location, ResourceBundle resources) {
+        List<Market> markets = model.getMarkets();
+
+        // Create a pane and set its properties
+        gridPane.setAlignment(Pos.CENTER);
+        gridPane.setPadding(new Insets(8, 0, 0, 0));
+        gridPane.setVgap(64);
+        gridPane.setHgap(64);
+
+        // Place nodes in the pane
+        for(int i = 1; i <= markets.size()/2; i++) {
+            gridPane.add(createMarketButton(i , markets),0, i - 1);
+        }
+        for(int i = markets.size()/2 + 1; i <= markets.size(); i++) {
+            gridPane.add(createMarketButton(i, markets),1, i - markets.size()/2 - 1);
+        }
```

Comment:
You can add all buttons in one loop

Reply:
for(int i = 0; i < markets.size(); i++) {
            if(i < markets.size()/2)
                gridPane.add(createMarketButton(markets.get(i)),0, i);
            else
                gridPane.add(createMarketButton(markets.get(i)),1, i - markets.size()/2);
        }